### PR TITLE
feat: Home button

### PR DESCRIPTION
**Pull request by vedovelli**
2022-02-01 11 53 50 jpg

We have a scroll to top button displayed on small screens. Now we also have the return home button, also for small screens only.

This implementation replaced ScrollToTop component with a new implementation comprising both cases.

## Rules
1. Should be displayed on small screens only
2. Should NOT be displayed when user is in the main route ("/")

## Testing

Integration tests are in place to assert all the cases. Just run `yarn test`.

### CODE REVIEW

File: web/app/components/Links/Internal/Internal.tsx [hunk 1]
```diff
@@ -7,9 +7,9 @@ export interface InternalProps {
   className?: string;
 }
 
-export function Internal({ href, children, className }: InternalProps) {
+export function Internal({ href, children, className, ...props }: InternalProps) {
   return (
-    <Link prefetch="intent" to={href} className={className}>
+    <Link prefetch="intent" to={href} className={className} {...props}>
```
Comment (vedovelli) [inline web/app/components/Links/Internal/Internal.tsx:12]: **dev note**: this allows us to pass `data-testid` to the component.

File: web/app/components/MobileNavigationItem/MobileNavigationItem.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      enterTo="opacity-100"
+      leave="transition-opacity duration-150"
+      leaveFrom="opacity-100"
+      leaveTo="opacity-0"
+      className={classNames(
+        'visible fixed h-16 w-16 bg-white  bg-opacity-40  shadow-md lg:invisible',
+        {
+          '-bottom-3 -right-3 rounded-tl-full': variant === Variant.ScrollToTop,
+          '-bottom-3 -left-3 rounded-tr-full': variant === Variant.Home,
+          'invisible': variant === Variant.Home && location.pathname === '/',
```
Comment (vedovelli) [inline web/app/components/MobileNavigationItem/MobileNavigationItem.tsx:49]: **dev note**: it should NOT be displayed when user is at Home.

File: web/app/features/Blog/components/Card/Card.tsx [hunk 3]
```diff
@@ -43,10 +43,11 @@ export function Card({ post, excerpt, picture, small = false }: CardProps) {
           href={route('/:slug', {
             slug: slug.current,
           })}
+          data-testid="post-link"
```
Comment (vedovelli) [inline web/app/features/Blog/components/Card/Card.tsx:46]: **dev note**: this is the only relevant change. Everything else is just Tailwind ESLint doing its job.

File: web/app/components/MobileNavigationItem/MobileNavigationItem.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  ScrollToTop,
+  Home,
+}
+
+export interface MobileNavigationItemProps {
+  variant: Variant;
+}
+
+export function MobileNavigationItem({ variant }: MobileNavigationItemProps) {
+  const [visible, setvisible] = useState<boolean>(false);
```
Comment (siimaoo) [inline web/app/components/MobileNavigationItem/MobileNavigationItem.tsx:17]: Isn't necessary pass type in this useState, as you pass a value, TS will infer a type as Boolean.

File: web/app/components/MobileNavigationItem/MobileNavigationItem.tsx (hunk 2 above)
Comment (siimaoo) [reply web/app/components/MobileNavigationItem/MobileNavigationItem.tsx:49]: Maybe separate this condition in a const will improve legibility

**Review by siimaoo — DISMISSED** [review]

**Review by siimaoo — APPROVED** [review]

**Review by siimaoo — APPROVED** [review]

**Pull request merged**